2006-07-20  Claudio Saavedra  <csaavedra@alumnos.utalca.cl>

* metadata/rb-metadata-dbus-client.c: (kill_metadata_service):
* metadata/rb-metadata-dbus-service.c: (main): Replace calls to 
deprecated dbus_connection_disconnect () with dbus_connection_close ().
Fixes #348098.

Index: metadata/rb-metadata-dbus-service.c
===================================================================
RCS file: /cvs/gnome/rhythmbox/metadata/rb-metadata-dbus-service.c,v
retrieving revision 1.10
retrieving revision 1.11
diff -u -r1.10 -r1.11
--- metadata/rb-metadata-dbus-service.c	14 Jul 2006 12:27:15 -0000	1.10
+++ metadata/rb-metadata-dbus-service.c	21 Jul 2006 04:36:16 -0000	1.11
@@ -501,7 +501,11 @@
 	g_main_loop_run (svc.loop);
 
 	if (svc.connection) {
+#ifdef WITH_OLD_DBUS
 		dbus_connection_disconnect (svc.connection);
+#else
+		dbus_connection_close (svc.connection);
+#endif
 		dbus_connection_unref (svc.connection);
 	}
 
Index: metadata/rb-metadata-dbus-client.c
===================================================================
RCS file: /cvs/gnome/rhythmbox/metadata/rb-metadata-dbus-client.c,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- metadata/rb-metadata-dbus-client.c	14 Jul 2006 12:27:15 -0000	1.7
+++ metadata/rb-metadata-dbus-client.c	21 Jul 2006 04:36:16 -0000	1.8
@@ -119,7 +119,11 @@
 	if (dbus_connection) {
 		if (dbus_connection_get_is_connected (dbus_connection)) {
 			rb_debug ("closing dbus connection");
+#ifdef WITH_OLD_DBUS
 			dbus_connection_disconnect (dbus_connection);
+#else
+			dbus_connection_close (dbus_connection);
+#endif
 		} else {
 			rb_debug ("dbus connection already closed");
 		}
